packages/cli/src/admin-sdk.ts

```diff
@@ -0,0 +1,224 @@
+import { createClient } from './autogen/client/createClient';
+import fse from 'fs-extra';
+import { kebabCase, omit } from 'lodash';
+import chalk from 'chalk';
+import { readAsJson, getFiles, getDirectories, replaceField } from './utils';
+import cliProgress from 'cli-progress';
+import { createHash } from 'crypto';
+import traverse from 'traverse';
+
+const multibar = new cliProgress.MultiBar(
+  {
+    clearOnComplete: false,
+    hideCursor: true,
+    format: '|{bar}| {name} | {value}/{total}',
+  },
+  cliProgress.Presets.shades_grey
+);
+
+// todo change this to prod
+const root = 'https://qa.builder.io';
+
+export const importSpace = async (privateKey: string, directory: string, debug = false) => {
+  const graphqlClient = createClient({
+    fetcher: ({ query, variables }, fetch, qs) =>
+      fetch(`${root}/api/v2/admin?${qs.stringify({ query, variables })}`, {
+        headers: {
+          'Content-Type': 'application/json',
+          Authorization: `Bearer ${privateKey}`,
+        },
+      }).then(r => r.json()),
+  });
+
+  const spaceProgress = multibar.create(1, 0);
+  spaceProgress.start(1, 0, { name: 'getting space settings' });
+
+  try {
+    const space = await graphqlClient.chain.query.downloadClone().execute({
+      models: { name: true, everything: true, content: true },
+      settings: true,
+      meta: true,
+    });
+    spaceProgress.setTotal(space.models.length);
+    await fse.outputFile(
+      `${directory}/settings.json`,
+      JSON.stringify({ ...space.settings, cloneInfo: space.meta }, undefined, 2)
+    );
+    const modelOps = space.models.map(async (model, index) => {
+      const { content, everything } = model;
+      // todo why conent is in everything
+      const { content: _, ...schema } = everything;
+      const modelName = kebabCase(model.name);
+      await fse.emptyDir(`${directory}/${modelName}`);
+      const modelProgress = multibar.create(content.length, 0, { name: modelName });
+      if (content.length > 0) {
+        modelProgress.start(content.length, 0);
+      }
+      await fse.outputFile(
+        `${directory}/${modelName}/schema.model.json`,
+        JSON.stringify(schema, null, 2)
+      );
+      await Promise.all(
+        content.map(async (entry, index) => {
+          const filename = `${directory}/${modelName}/${kebabCase(entry.name)}.json`;
+          await fse.outputFile(filename, JSON.stringify(entry, undefined, 2));
+          modelProgress.increment(1, { name: ` ${modelName}: ${filename} ` });
+        })
+      );
+      spaceProgress.increment();
+      modelProgress.stop();
+    });
+    await Promise.all(modelOps);
+    if (debug) {
+      console.log(chalk.green('Imported successfully ', space.settings.name));
+    }
+  } catch (e) {
+    console.log(`\r\n\r\n`);
+    console.error(chalk.red('Error importing space'));
+    console.error(e);
+    process.exit();
+  }
```

Comment:
Since the only thing we do here is report out the error, I suggest to not use a try-catch here, and have the caller report the problem. The caller could (optionally) pass in the progress bar / multibar instances, allowing the inner function to still report progress.

```
function importSpace(privateKey: string, directory: string, logger?, progressBar?) { ... }

// CLI
  // ...
  .action(options => {
    try {
       importSpace(options.key, options.output, loggerInstance, progressBarInstance);
    } catch (err) {
      console.error(`\r\n\r\n`, chalk.red('Error importing space', err));
      process.exit(1);
    }
  }); 
```

This simplifies the actual worker functions a lot, by them not having to worry too much about error handling and user reporting, that's the responsibility of the CLI layer.